### PR TITLE
First PR for cabal-2.0 support

### PR DESCRIPTION
**Pull request by erikd-ambiata**
Hacking this to make it work made a rather large mess. Going to split this up as a set of smaller PRs.

! @jystic @tmcgilchrist 
/jury approved @jystic

### CODE REVIEW

**Comment by erikd-ambiata** [comment]
! @jystic @tmcgilchrist This PR only changes the package plan parser so that it is *capable* of parsing the cabal-2.0 package plan output. Other than changing the this parser it has no other functional changes.

File: src/Mafia/Cabal/Dependencies.hs [hunk 1]
```diff
@@ -293,9 +304,18 @@ pPackageChange =
...
+pPackageId :: Parser PackageId
+pPackageId = do
+  let
+    isPkgIdChar c
+        | c >= 'a' && c <= 'z' = True
+        | c >= 'A' && c <= 'Z' = True
+        | c >= '0' && c <= '9' = True
+        | c == '-' = True
+        | c == '.' = True
+        | otherwise = False
```
Comment (jacobstanley) [inline src/Mafia/Cabal/Dependencies.hs:316]: Curious where you found these rules? (just curious)

**Comment by jacobstanley** [comment]
👍 looks ok to me, I'm assuming you've tested this out with the cabal 1.24 and everything still works

File: src/Mafia/Cabal/Dependencies.hs (hunk 1 above)
Comment (erikd-ambiata) [reply src/Mafia/Cabal/Dependencies.hs:316]: Its from memory. I read it in the cabal sources at some point. At the moment google can only find : https://github.com/commercialhaskell/stack/blob/master/src/Stack/Types/PackageName.hs#L76-L84

**Comment by erikd-ambiata** [comment]
> I'm assuming you've tested this out with the cabal 1.24 and everything still works

Yep, I was running mafia with this and some other patches and then switching back and forth between cabal-install-1.24 and 2.0 while I was fiddling with https://github.com/ambiata/cabal-extras/pull/68 and https://github.com/ambiata/parsley/pull/37 .

**Comment by tmcgilchrist** [comment]
Nice work!

**Pull request merged**